### ISSUE TITLE
Trouble rendering style color palettes in various projections using wms in cartopy

### ISSUE THREAD

**Issue by hevgyrt** [SciTools/cartopy#1386]
### Description

I try to plot data from wms using cartopy. In particular, I would like to both change color style and projection for some applications. I am able to change style using PlateCaree as projection. However, when I try other projections in ccrs like Mercator() the color palette is always gray scale. Am I doing something wrong?

#### Code to reproduce

```
import cartopy.crs as ccrs
import cartopy
import matplotlib.pyplot as plt
from owslib.wms import WebMapService

url = "http://thredds.met.no/thredds/wms/sea/arctic20km/1h/aggregate_be?service=WMS&version=1.3.0&request=GetCapabilities"
wms = WebMapService(url,timeout=80)

layer='temperature'
style = ['boxfill/redblue']

projections = [ccrs.Mercator(), ccrs.PlateCarree()]
for projection in projections:
    subplot_kw = dict(projection=projection)
    fig, ax = plt.subplots(subplot_kw=subplot_kw)
    
    ax.add_wms(wms, layer,
	       wms_kwargs={'transparent': False,
               'styles':style})
    print(wms.contents[layer].styles[style[0]])
    plt.show()


```

#### Traceback 

```

```

<details>
 <summary>Full environment definition</summary>

### Operating system
Linux ubuntu 18.04

### Cartopy version
'0.17.0'

### conda list

```
```

### pip list

```
```

</details>

**Comment by htonchia** [comment]
You did nothing wrong.

Well, it seems that when the image sent by the server is not an RGB(A) image, the color palette is lost in the regrid process to the target projection.

If one convert the image from the server to RGBA, one gets:
![MercatorBlueRed](https://user-images.githubusercontent.com/54799451/70455711-0b54e980-1aad-11ea-8136-92db1c4188ba.png)
I guess it is what you expected.

it is fixed by:         
```
# convert to RGBA to keep the color palette
        img, extent = warp_array(np.asanyarray(image.convert('RGBA'))[::-1]
```
in io/ogc_clients.py

**Comment by greglucas** [comment]
Closed by #1473
